src/ase/cw/model/Server.java

```diff
@@ -81,12 +91,27 @@ public void startOrderProcess() {
         } else {
             System.out.println(getName() + "already started");
         }
+
+    }
+
+    @Override
+    public void pauseOrderProcess() {
+        // this does not pause the status immediately but waits until current order is processed
+        serverStatus = status.PAUSED;
```

Comment:
Furthermore: is you change the serverStatus without useint the setter, ssl.onServerStatusChange(this); is not called

Reply:
this is correct for pause action. the setter is not called until the order is processed so serverStatus is only used to notify the run method to call the setter once order is done

Reply:
okay, in this case a helper variable could be usefull.
like:
bollean shouldPause =false;
inside pauseOrderProcess()  you set shouldPause =true;
and inside run() you can check if(shouldPause )  setState(pause); wait();

So you seperate between the state you want, the current state. 

But if you implement this right your solution should also work^^

Reply:
I have added TO_BE_PAUSED status in my last commit